File: PracticasProgramacion/Practica_16/cpp/Fractales.cpp
#include "Fractales.h"
#include <iostream>
#include <cmath>

void Fractales::sierpinskiAux(int nivel, const std::string& prefijo) const {
    if (nivel <= 0) return;
    std::cout << prefijo << "  *\n";
    std::cout << prefijo << " ***\n";
    std::cout << prefijo << "*****\n\n";
    sierpinskiAux(nivel - 1, prefijo + "   ");
}

void Fractales::trianguloSierpinski(int nivel) const {
    std::cout << "Triangulo Sierpinski nivel " << nivel << ":\n";
    for (int i = 0; i < nivel; ++i) {
        std::string esp(i * 2, ' ');
        std::cout << esp << "  *\n" << esp << " ***\n" << esp << "*****\n";
    }
}

void Fractales::polvoCantor(int nivel, int indent) const {
    if (nivel <= 0) {
        std::cout << std::string(indent, ' ') << "___" << std::endl;
        return;
    }
    polvoCantor(nivel - 1, indent);
    std::cout << std::string(indent, ' ') << "   " << std::endl;
    polvoCantor(nivel - 1, indent + 4);
}

File: PracticasProgramacion/Practica_16/python/fractales.py
"""
Fractales: triángulo de Sierpinski (repetición por nivel) y polvo de Cantor recursivo.
Cantor: nivel 0 dibuja segmento; nivel n divide en dos mitades con hueco central.
"""


class Fractales:
    def triangulo_sierpinski(self, nivel: int) -> None:
        print(f"Triangulo Sierpinski nivel {nivel}:")
        for i in range(nivel):
            esp = " " * (i * 2)
            print(f"{esp}  *")
            print(f"{esp} ***")
            print(f"{esp}*****")

    def polvo_cantor(self, nivel: int, indent: int) -> None:
        if nivel <= 0:
            print(" " * indent + "___")
            return
        self.polvo_cantor(nivel - 1, indent)
        print(" " * indent + "   ")
        self.polvo_cantor(nivel - 1, indent + 4)
